app-arch/tar/patches/tar-1.29.patchset

```diff
@@ -5,26 +5,23 @@ Subject: gcc2 patch
 
 
 diff --git a/lib/wordsplit.h b/lib/wordsplit.h
-index d64cf2f..c93f53d 100644
+index e0d2559..acf1f56 100644
 --- a/lib/wordsplit.h
 +++ b/lib/wordsplit.h
-@@ -34,9 +34,9 @@ struct wordsplit
-   const char *ws_escape;
-   void (*ws_alloc_die) (struct wordsplit * wsp);
-   void (*ws_error) (const char *, ...)
--                   __attribute__ ((__format__ (__printf__, 1, 2)));
-+;
-   void (*ws_debug) (const char *, ...)
--                   __attribute__ ((__format__ (__printf__, 1, 2)));
-+;
+@@ -22,7 +22,7 @@
  
-   const char **ws_env;
-   const char *(*ws_getvar) (const char *, size_t, void *);
+ #include <stddef.h>
+ 
+-#if 2 < __GNUC__ + (7 <= __GNUC_MINOR__)
++#if 2 < __GNUC__ + (7 <= __GNUC_MINOR__) - __HAIKU__
```

Comment:
Sorry, I just realized that you were not asking for an explanation __about__ the hack but for an explanation __in__ the patch header.
BTW, your comment makes me think it would be even better to replace the **`(7 <= __GNUC_MINOR__)`** by **`(95 < __GNUC_MINOR__)`** and not use `__HAIKU__` at all.
This would probably be upstreamable because what I noticed is that the **`__attribute__ ((__format__ spec))`** does not work with our gcc 2.95. Well, we would need to double check why upstream thinks gcc >= 2.7 supports **`__attribute__`**.
Shall I change the patch in that sense? I mean replacing "**`7 <=`**" by "**`95 < `**".
May I let you dig that question, since you know gcc **much** better than me?
Thanks!